### PR TITLE
feat:(KW-142) Allow end user to search for content in the map

### PR DESCRIPTION
**Pull request by Nashmod**
### Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### How others should test

_Does this need to be tested by QA in the next release cycle? If so please give a brief explanation of how to test these changes._

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: src/components/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        vendor.title.toLowerCase().includes(search.toLowerCase()) ||
+        _.some(vendor.tags, (tag) => tag.toLowerCase().includes(search.toLowerCase()))
+    )
+    dispatch(setFilteredVendors(filteredVendors))
+  }, [search])
+
+  const handleSearch = (search: string) => {
+    setSearch(search)
+    dispatch(setSearchQuery(search))
+  }
```
Comment (shottah) [inline src/components/SearchBar.tsx]: This logic should be triggered by `vendor/saga.ts`

E.g. `takeEvery(Action.SET_SEARCH_QUERY, watchMapFilter)`
```ts
function* watchMapFilter (action: any) {
  const vendors = filterVendors(action.payload)
  yield put(setFilteredVendors(vendors))
}
```

Just take note of this pattern. With this, you have taken the logic and decoupled it from the component that handles searching.

File: src/vendors/actions.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
-import { Vendor, Vendors } from 'src/vendors/types'
+import { Vendor, Vendors, VendorWithLocation } from 'src/vendors/types'
 
 export enum Actions {
   FETCH_VENDORS = 'VENDORS/FETCH_VENDORS',
   SET_VENDORS = 'VENDORS/SET_VENDORS',
   SET_LOADING = 'VENDORS/SET_LOADING',
   SET_CURRENT_VENDOR = 'VENDORS/SET_CURRENT_VENDOR',
+  SET_FILTERED_VENDORS = 'VENDORS/SET_FILTERED_VENDORS',
+  SET_SEARCH_QUERY = 'VENDORS/SET_SEARCH_QUERY',
```
Comment (shottah) [inline src/vendors/actions.ts]: This search bar is used for the entire map, not just vendors. I would create a new reducer for `map` general actions like this.

See the updates. The new map reducer is here.

**Review by shottah — COMMENTED** [review]
I will be merging an update into this branch for you to continue.

See the changes merged in from #67

**Comment by shottah** [comment]
See this screenshot for my changes.
https://user-images.githubusercontent.com/14793093/193723639-63be1699-d53c-4771-9ec7-cce280ba37bd.mp4

File: src/components/SearchBar.tsx (hunk 1 above)
Comment (shottah) [reply src/components/SearchBar.tsx]: Also please refrain from importing the `_` namespace from `lodash`. Only import the functions you need.

E.g. `import { some } from 'lodash'`

**Review by shottah — APPROVED** [review]
Will need code in the future to handle:
 * Focusing search should open the map sheet
 * Focusing a vendor should hide the search bar
..and other state related interactions like this.

**Comment by shottah** [comment]
@Nashmod ready to merge.

**Comment by Nashmod** [comment]
> Will need code in the future to handle:
> 
> * Focusing search should open the map sheet
> * Focusing a vendor should hide the search bar
>   ..and other state related interactions like this.

Agreed!

**Pull request merged**